Quantum-Control-Applications/ESR/T1.py

```diff
@@ -0,0 +1,172 @@
+from qm import SimulationConfig
+from qm.qua import *
+from qm import LoopbackInterface
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+from qm.simulate.credentials import create_credentials
+
+
+###################
+# The QUA program #
+###################
+
+pi_len = 320 // 4
+pi_half_len = 160 // 4
+
+wait_min = 1500 // 4
+wait_max = 20000 // 4
+dwait = 1000 // 4
+
+waits = np.arange(wait_min, wait_max + 0.1, dwait)
+
+cooldown_time = int(10e6 // 4)
+
+n_avg = 100
+
+pulse_delay_p = int(2000 // 4 - pi_half_len * 0.5 - pi_len * 0.5)
+readout_delay_p = int(2000 // 4 - pi_len * 0.5 - readout_len * 0.125 - 5)
+
+with program() as T1:
+
+    n = declare(int)
+    wait_len = declare(int)
+    pulse_delay = declare(int, value=pulse_delay_p)
+    readout_delay = declare(int, value=readout_delay_p)
+
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+
+        with for_(wait_len, wait_min, wait_len <= wait_max, wait_len + dwait):
+
+            # initialization
+            # wait(wait_len)
+
+            reset_phase("ensemble")
+            reset_phase("resonator")
+            reset_frame("ensemble")
+
+            echo = declare_stream(adc_trace=True)
+
+            wait(8, "switch_1", "switch_2")
+            play("activate", "switch_1", duration=pi_half_len)
+            play("activate", "switch_2", duration=pi_half_len)
+            play("const", "ensemble", duration=pi_half_len)
+
+            wait(pulse_delay, "ensemble", "switch_1", "switch_2")
+
+            frame_rotation_2pi(-0.5, "ensemble")
+            play("activate", "switch_1", duration=pi_len)
+            play("activate", "switch_2", duration=pi_len)
+            play("const", "ensemble", duration=pi_len)
+
```

Comment:
I am not sure I understand the sequence, you play first pi/2 and then after waiting some fixed time you play pi, then wait again and measure. Why this sequence of pi/2 and pi and why don't you use wait_len?

Reply:
Ok, thanks I'll definitely need to read about these techniques before the next NV installation...
However it seems that the duration in the wait statements is not iterating, is it normal? How do you get T1 if the time is not scanned?

Reply:
Oh yes got it thanks !